src/ConnectedMode/ServerSentEvents/SSESessionFactory.cs

```diff
@@ -0,0 +1,183 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.ServerSentEvents.Issues;
+using SonarLint.VisualStudio.Core.ServerSentEvents.TaintVulnerabilities;
+using SonarQube.Client;
+using SonarQube.Client.Models.ServerSentEvents.ClientContract;
+
+namespace SonarLint.VisualStudio.ConnectedMode.ServerSentEvents
+{
+    /// <summary>
+    /// Factory for <see cref="ISSESession"/>. Responsible for disposing EventSourcePublishers
+    /// </summary>
+    internal interface ISSESessionFactory : IDisposable
+    {
+        ISSESession Create(string projectKey);
+    }
+
+    /// <summary>
+    /// Represents the session entity, that is responsible for dealing with SQ Client's SSE Session reader
```

Comment:
Minor: why not a cref?

Reply:
I didn't have the interface at that time. But now that you pulled it I can fix this